calour/io.py

```diff
@@ -80,13 +84,50 @@ def _read_biom(fp, transpose=True):
     return sid, fid, data, feature_md
 
 
-def _read_qiime2(fp, transpose=True):
-    '''Read in a qiime2 .qza biom table file.
+def _file_from_zip(tempdir, data_file, internal_data):
+    '''extract the data file from a regular/qza filename into the tempdir, and return the filename
+
+    Parameters
+    ----------
+    tmpdir: str
+        name of the directory to extract the zip into
+    data_file: str
+        original name of the file (could be '.qza' or not)
+    internale_data: str
+        the internal qiime2 qza file name (i.e. 'data/feature-table.biom' for biom table etc.)
+
+    Returns
+    -------
+    str: name of data file inside the tempdir
+    '''
+    if not zipfile.is_zipfile(data_file):
+        raise ValueError('the qiime2 file %s is not a valid zip file. Is it a qiime2 artifact (.qza) file?' % data_file)
+    fl = zipfile.ZipFile(data_file)
+    internal_name = None
+    for fname in fl.namelist():
+        if fname.endswith(internal_data):
+            internal_name = fname
+            break
+    if internal_name is None:
+        raise ValueError('No data file (%s) in qza file %s. is it the appropriate qiime2 file?' % (internal_data, data_file))
+    data_file = fl.extract(internal_name, tempdir)
+    return data_file
+
+
+def _read_qiime2_zip(fp, transpose=True):
+    '''Read in a qiime2 qza biom table with optional rep_seq and taxonomy data files
+    NOTE: this function reads2 the qiime2 qza artifacts files using unzip rather than relying on qiime2.
+    This enables reading qiime2 artifacts without activating the qiime2 environment
 
     Parameters
     ----------
     fp : str
         file path to the qiime2 (.qza) biom table
+    rep_seq_file: str or None, optional
+        If not none, name of the representative sequencees qiime2 artifact, used to translate sequnece hashes
+        to the actual sequences.
+    taxonomy_file: str or None, optional
+        If not none, name of the taxonomy qiime2 artifact file.
```

Comment:
these parameters are not in the function API

Reply:
oops. fixed